### DISCUSSION TITLE
Displaying image next to the code?

### DISCUSSION THREAD

**Discussion by weirane** [microsoft/vscode-discussions#280] [Extension Development QnA] [answered]
Hi, I'm wondering if it's possible for an extension to display an SVG image next to the code? I want to display an image to aid the comprehension of error message and I need to align the image with the code.

**Comment by alefragnani** [comment, marked as answer]
Hi @weirane ,

Yes, you can. Simply use the `createTextEditorDecorationType` API. A sample is provided here https://github.com/microsoft/vscode-extension-samples/tree/main/decorator-sample.

I'm not sure your requirements (image size for instance), but one of the extensions I remember do this is Jumpy (https://marketplace.visualstudio.com/items?itemName=wmaurer.vscode-jumpy). And even the native color picker may be something you would have.

Hope this helps

**Reply by weirane** [reply]
Thanks for the answer. Will the decorator be on top of the text or can I specify a position that is to the right of the text? I don't want the image and the code to overlay.

**Reply by alefragnani** [reply]
You can choose the location, as the API provides you some flexibility.

Look for `vscode.window.createTextEditorDecorationType` and its options https://code.visualstudio.com/api/references/vscode-api#DecorationRenderOptions

**Reply by weirane** [reply]
Is there a way to display the image across multiple lines? It seems like the decoration is just for one line. I specified the range as `range: new vscode.Range(0, 0, 2, 40),` but it seems like the decoration can only be displayed at line 3.

<img width="516" alt="2022-12-12_11-03" src="https://user-images.githubusercontent.com/37568244/207132229-07424c84-6c5c-4072-8387-f83cb727938b.png">

This is my `activate` function:
```typescript
export function activate(context: vscode.ExtensionContext) {
  const dtype = vscode.window.createTextEditorDecorationType({
    isWholeLine: true,
    after: {
      height: '0',
    },
  });
  const options = {
    range: new vscode.Range(0, 0, 2, 40),
    renderOptions: {
      dark: {
        after: {
          contentIconPath: vscode.Uri.parse('file:///path/to/vis_timeline.svg'),
        }
      },
      light: {
        after: {
          contentIconPath: vscode.Uri.parse('file:///path/to/vis_timeline.svg'),
        }
      },
    },
  };
  const editor = vscode.window.activeTextEditor;
  if (!editor) {
    console.log('no editor');
    return;
  }
  editor.setDecorations(dtype, [options]);
}
```
